base/src/types.rs

```diff
@@ -533,6 +566,17 @@ impl<Id, T> Type<Id, T>
     }
 }
 
+pub trait TypeRef: Sized {
```

Comment:
You could also just implement it as a free function. I guess I am just wary of traitifying too early if it is not yet needed, and the right abstractions are not yet clear.